### PR TITLE
Change DeployerService interface to accept Deployment object

### LINKED ISSUE #69: Refactor DeployerService interface
**Description:**
Refactor DeployerService interface and it's associated, model objects to accommodate the location of deployment scripts.

**Suggested Assignees:**
harshanl

**Related Issues:**
#68

### PR DESCRIPTION
**Pull request by harshanL**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: core/src/test/java/org/wso2/carbon/testgrid/core/TestGridMgtServiceTest.java [hunk 1]
```diff
@@ -155,34 +158,32 @@ public void executeProductTestPlanTest() throws TestGridException, UnsupportedPr
...
+            PowerMockito.whenNew(DeployerServiceImpl.class).withNoArguments().thenReturn(deployer);
+            PowerMockito.whenNew(TestEngineImpl.class).withNoArguments().thenReturn(testEngine);
+            PowerMockito.whenNew(TestReportEngineImpl.class).withNoArguments().thenReturn(testReportEngine);
         } catch (Exception e) {
             e.printStackTrace();
         }
 
 
-        testGridMgtService.executeProductTestPlan(productTestPlan);
+        new TestGridMgtServiceImpl().executeProductTestPlan(productTestPlan);
```
Comment (vidurananayakkara) [inline core/src/test/java/org/wso2/carbon/testgrid/core/TestGridMgtServiceTest.java]: Remove extra blank line

File: core/src/test/java/org/wso2/carbon/testgrid/core/TestGridMgtServiceTest.java [hunk 2]
```diff
@@ -155,34 +158,32 @@ public void executeProductTestPlanTest() throws TestGridException, UnsupportedPr
...
-            whenNew(DeployerServiceImpl.class).withNoArguments().thenReturn(deployer);
-            Mockito.when(deployer.deploy(testPlan)).thenReturn(deployment);
-
-            whenNew(TestEngineImpl.class).withNoArguments().thenReturn(testEngine);
-            Mockito.when(testEngine.runScenario(testScenario, scenarioLocation, deployment)).thenReturn(true);
-
-            whenNew(TestReportEngineImpl.class).withNoArguments().thenReturn(testReportEngine);
+            PowerMockito.whenNew(DeployerServiceImpl.class).withNoArguments().thenReturn(deployer);
+            PowerMockito.whenNew(TestEngineImpl.class).withNoArguments().thenReturn(testEngine);
+            PowerMockito.whenNew(TestReportEngineImpl.class).withNoArguments().thenReturn(testReportEngine);
```
Comment (vidurananayakkara) [inline core/src/test/java/org/wso2/carbon/testgrid/core/TestGridMgtServiceTest.java]: Why are we catching the exception? If the exception is thrown it means that the test should fail right?

File: common/src/main/java/org/wso2/carbon/testgrid/common/constants/TestGridConstants.java [hunk 3]
```diff
@@ -13,4 +13,6 @@
     public static final String TEST_TYPE_TESTNG = "TESTNG";
 
     public static final String JMTER_SUFFIX = ".jmx";
+
```
Comment (vidurananayakkara) [inline common/src/main/java/org/wso2/carbon/testgrid/common/constants/TestGridConstants.java]: Remove the additional blank line here

File: core/src/test/java/org/wso2/carbon/testgrid/core/TestGridMgtServiceTest.java (hunk 1 above)
Comment (harshanL) [reply core/src/test/java/org/wso2/carbon/testgrid/core/TestGridMgtServiceTest.java]: Will fix with #54

File: core/src/test/java/org/wso2/carbon/testgrid/core/TestGridMgtServiceTest.java (hunk 2 above)
Comment (harshanL) [reply core/src/test/java/org/wso2/carbon/testgrid/core/TestGridMgtServiceTest.java]: Will fix with #54

File: common/src/main/java/org/wso2/carbon/testgrid/common/constants/TestGridConstants.java (hunk 3 above)
Comment (harshanL) [reply common/src/main/java/org/wso2/carbon/testgrid/common/constants/TestGridConstants.java]: Fixed with 69b668b93159143b0eb766bf2210e1ad31f5f5e7

**Pull request merged**